thirdeye-server/src/main/java/ai/startree/thirdeye/auth/AuthorizationManager.java

```diff
@@ -129,13 +141,54 @@ public boolean hasRootAccess(final ThirdEyePrincipal principal) {
         accessControl.hasAccess(principal.getAuthToken(), ROOT_RESOURCE_ID, AccessType.WRITE);
   }
 
-  static public <T extends AbstractDTO> ResourceIdentifier resourceId(final T dto) {
-    return ResourceIdentifier.from(
-        optional(dto.getId()).map(Objects::toString).orElse(DEFAULT_NAME),
-        optional(dto.getAuth())
-            .map(AuthorizationConfigurationDTO::getNamespace).orElse(DEFAULT_NAMESPACE),
-        optional(SubEntities.BEAN_TYPE_MAP.get(dto.getClass()))
-            .map(Objects::toString).orElse(DEFAULT_ENTITY_TYPE));
+  // Returns the resource identifier for a dto.
+  // Null is ok and maps to a default resource id.
+  public ResourceIdentifier resourceId(final AbstractDTO dto) {
+    final var name = optional(dto)
+        .map(AbstractDTO::getId)
+        .map(Objects::toString)
+        .orElse(DEFAULT_NAME);
+
+    final var namespace = optional(dto)
+        .map(this::resolveAuthParentDto)
+        .map(AbstractDTO::getAuth)
+        .map(AuthorizationConfigurationDTO::getNamespace)
+        .orElse(DEFAULT_NAMESPACE);
+
+    final var entityType = optional(dto)
+        .map(AbstractDTO::getClass)
+        .map(SubEntities.BEAN_TYPE_MAP::get)
+        .map(Objects::toString)
+        .orElse(DEFAULT_ENTITY_TYPE);
+
+    return ResourceIdentifier.from(name, namespace, entityType);
+  }
+
+  // Resolves the Authorization parent for certain dto classes.
+  // AnomalyDTO -> EnumerationItemDTO if exists, else AlertDTO
+  // Everything else -> itself
+  private AbstractDTO resolveAuthParentDto(final AbstractDTO dto) {
```

Comment:
Yeah this is completely my bad. we need an inheritance hierarchy here. Let's come back to this later. This is ok for now.